providers/implementations/exchange/dh_exch.c

```diff
@@ -390,9 +395,16 @@ static int dh_set_ctx_params(void *vpdhctx, const OSSL_PARAM params[])
     p = OSSL_PARAM_locate_const(params, OSSL_KDF_PARAM_CEK_ALG);
     if (p != NULL) {
         str = name;
-        if (!OSSL_PARAM_get_utf8_string(p, &str, sizeof(name)))
-            return 0;
-        pdhctx->kdf_cekalg = OPENSSL_strdup(name);
+
+        OPENSSL_free(pdhctx->kdf_cekalg);
+        pdhctx->kdf_cekalg = NULL;
+        if (p->data != NULL && p->data_size != 0) {
```

Comment:
I think `OSSL_PARAM_get_utf8_string()` does these checks for you.

Reply:
> It also produces errors on the error stack.

Yeah, we would have to clear them. So that won't be much better than this solution.